src/prog_models/sim_result.py

```diff
@@ -165,17 +173,35 @@ def pop(self, index : int = -1):
             return self.__data.pop(index)
         return self.fcn(x)
 
-    def remove(self, target) -> None:
+    def remove(self, d = None, t = None, s = None) -> None:
         """Remove an element
-
+         
         Args:
-            target: Value of element to be removed.
-        """
-        target_index = self.data.index(target)
-        self.times.pop(target_index)
-        self.states.pop(target_index)
-        if self.__data is not None:
-            self.__data.remove(target)
+            d: Data value to be removed.
+            t: Time value to be removed.
+            s: State value to be removed.
+        """ 
+        if (not [i for i in (d,t,s) if i is not None or None]) or (sum([i is None for i in (d, t, s)]) != 2):
+            raise ValueError("ValueError: Only one named argument (d, t, s) can be specified.")
+       
+        if (t is not None):
+            target_index = self.times.index(t)
+            self.times.pop(target_index)
+            self.states.pop(target_index)
+            if self.__data is not None:
+                self.__data.pop(target_index)
+        elif (s is not None):
+            target_index = self.states.index(s)
+            self.times.pop(target_index)
+            self.states.pop(target_index)
+            if self.__data is not None:
+                self.__data.pop(target_index)
+        else:
+            target_index = self.data.index(d)
+            self.times.pop(target_index)
+            self.states.pop(target_index)
+            if self.__data is not None:
```

Comment:
I think the if statement on Line 203 isn't actually necessary. __data is resolved on line 200 when you access it to check the index. Instead you can pop __data without checking if its none.